bindings/matrix-sdk-ffi/src/sliding_sync.rs

```diff
@@ -26,29 +26,46 @@ use crate::{
     TimelineListener,
 };
 
+type StoppableSpawnCallback = Box<dyn FnOnce() + Send + Sync>;
+
 pub struct StoppableSpawn {
-    handle: Arc<RwLock<Option<JoinHandle<()>>>>,
+    handle: JoinHandle<()>,
+    callback: RwLock<Option<StoppableSpawnCallback>>,
```

Comment:
Since `set_callback` is a `&mut self` method, this doesn't seem like it needs the `RwLock` either. (sorry could have been more explicit about this before)

Reply:
No, we need that unfortunately. Because `cancel(&self)` is called from FFI and can't be `&mut self` in which we remove the callback so it doesn't get called again. We need this to stay some mutex.